src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,143 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+/**
+ * Class for parsing user commands.
+ */
+public class Parser {
+    private static final String TODO_COMMAND = "todo";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String EVENT_COMMAND = "event";
+    private static final String LIST_COMMAND = "list";
+    private static final String DONE_COMMAND = "done";
+    private static final String FIND_COMMAND = "find";
+    private static final String DELETE_COMMAND = "delete";
+    private static final String EXIT_COMMAND = "bye";
+    private static final DateTimeFormatter dateTimeFormatter =
+            DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+
+    /**
+     * Parse user command as a String and returns the corresponding UserCommandType.
+     * @param command String of user command
+     * @return UserCommandType
+     * @throws InvalidCommandException if user command is not recognised
+     */
+    public static UserCommandType parseUserCommand(String command) throws InvalidCommandException {
+        if (command.equals(LIST_COMMAND)) {
+            return UserCommandType.LIST;
+        } else if (command.equals(EXIT_COMMAND)) {
+            return UserCommandType.EXIT;
+        } else if (command.startsWith(DONE_COMMAND)) {
+            return UserCommandType.DONE;
+        } else if (command.startsWith(DELETE_COMMAND)) {
+            return UserCommandType.DELETE;
+        } else if (command.startsWith(TODO_COMMAND)) {
+            return UserCommandType.TODO;
+        } else if (command.startsWith(DEADLINE_COMMAND)) {
+            return UserCommandType.DEADLINE;
+        } else if (command.startsWith(EVENT_COMMAND)) {
+            return UserCommandType.EVENT;
+        } else if (command.startsWith(FIND_COMMAND)) {
+            return UserCommandType.FIND;
+        } else {
+            throw new InvalidCommandException();
+        }
+    }
+
+    /**
+     * Parse a task command and return the String components.
+     * @param command user command String
+     * @return String array of the command components as Task name, dateTimeString
+     * @throws InvalidCommandException if command syntax is unrecognised
+     */
+    public static String[] parseTask(String command) throws InvalidCommandException {
+        String[] components = command.split(" ");
+        String taskType = components[0];
+
+        if (taskType.equals(TODO_COMMAND)) {
+            if (components.length < 2) {
+                throw new InvalidCommandException("The description of a todo cannot be empty.");
+            }
+            return new String[]{
+                    String.join(" ", Arrays.copyOfRange(components, 1, components.length))
```

Comment:
I'm not sure whether the layout here is correct... consider adding a blank between "]" and "{", consider changing the indentation to 4 characters instead of 8.